src/screens/nano/NanoContractExecuteMethod.js

```diff
@@ -0,0 +1,635 @@
+ * 
+ * In the future we should have a way to validate this.
+ *
+ * @memberof Screens
+ */
+function NanoContractExecuteMethod(props) {
+  const location = useLocation();
+  const data = location.state;
+  const isCreateNanoContract = data.method === hathorLib.constants.NANO_CONTRACTS_INITIALIZE_METHOD && data.ncId == null;
+  const {
+    nanoContracts,
+    blueprintsData,
+    tokenMetadata,
+    decimalPlaces,
+    registeredTokens,
+  } = useSelector((state) => {
+    return {
+      nanoContracts: state.nanoContracts,
+      blueprintsData: state.blueprintsData,
+      tokenMetadata: state.tokenMetadata,
+      decimalPlaces: state.serverInfo.decimalPlaces,
+      registeredTokens: state.tokens,
+    }
+  });
+
+  let addressToSign = null;
+  if (!isCreateNanoContract) {
+    // The nano must be registered in the wallet with an address
+    addressToSign = nanoContracts[data.ncId].address;
+  }
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const wallet = getGlobalWallet();
+
+  const formExecuteMethodRef = useRef(null);
+  const addressRef = useRef(null);
+  // List of refs of the arguments form
+  const formRefs = [];
+
+  // {string | null} If there is an error message when sending
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  // This will store each action in an array of objects
+  // { type, token, amount, address }
+  const [actions, setActions] = useState([]);
+  const globalModalContext = useContext(GlobalModalContext);
+
+  // We must store the ref of action addresses because we use for the validation
+  const actionAddressesRef = useRef([]);
+
+  /**
+   * Method executed when user clicks to execute the method
+   * We validate the arguments form and open the PIN modal
+   */
+  const executeMethod = () => {
+    let hasError = false;
+
+    /**
+     * Helper method to validate address and set custom validity in the refs
+     *
+     * @param {string} value Value from the address input
+     * @param {Object} ref Input reference to set custom validity
+     */
+    const validateAddress = (value, ref) => {
+      // Optional values were already handled
+      if (!value) {
+        hasError = true;
+        ref.setCustomValidity(t`Address is required.`);
+        return;
+      }
+
+      const addressObj = new hathorLib.Address(value, { network: wallet.getNetworkObject() });
+      if (addressObj.isValid()) {
+        ref.setCustomValidity('');
+        return;
+      }
+
+      ref.setCustomValidity(t`Invalid address.`);
+      hasError = true;
+    }
+
+    // The Address regex can be achieved but it's not good for readability or maintenance
+    // then I prefer to validate all Address inputs here
+    // First I validate the Address inputs in the arguments list, then I validate the addresses
+    // in the withdrawal actions
+    const args = get(data.blueprintInformation.public_methods, `${data.method}.args`, []);
+    for (let i=0; i<args.length; i++) {
```

Comment:
I'm not sure I agree with this change. I think it's clear to read the classic `for (let i=0; i<x; i++)` and the other way also has problemas. Even though for this case it has no downsides, I prefer using the code the would never have downsides. 

I usually use `for .. in` in javascript to iterate through objects, so I think it would be confusing to see it.